code/game/objects/items/reagent_containers/hypospray.dm

```diff
@@ -166,6 +118,74 @@
 
 	return TRUE
 
+///Always draws reagents on right click
+/obj/item/reagent_containers/hypospray/afterattack_alternate(atom/A, mob/living/user)
+	if(!in_range(A, user) || !user.Adjacent(A)) //So we arent drawing reagent from a container behind a window
+		return FALSE
+	can_draw_reagent(A, user, TRUE)
+
+///If it's possible to draw from something. Will draw_blood() when targetting a carbon, or draw_reagent() when targetting a non-carbon
+///Returns TRUE if we successfully draw from something
+/obj/item/reagent_containers/hypospray/proc/can_draw_reagent(atom/A, mob/living/user, alternate_draw_mode)
+	if(!A.reagents)
+		return FALSE
+	if(!istype(user))
+		return FALSE
+	if(!(inject_mode == HYPOSPRAY_INJECT_MODE_DRAW || alternate_draw_mode)) //If we aren't drawing or using right click, return FALSE.
+		return FALSE
```

Comment:
This should probs still be in the afterattack proc. That also means you don't need the alt_draw_mode arg.